base/mpfr.jl

```diff
@@ -1050,15 +1050,22 @@ _precision_with_base_2(::Type{BigFloat}) =
 """
     setprecision([T=BigFloat,] precision::Int; base=2)
 
-Set the precision (in bits, by default) to be used for `T` arithmetic.
+Set the global precision (in bits, by default) to be used for `T` arithmetic.
 If `base` is specified, then the precision is the minimum required to give
 at least `precision` digits in the given `base`.
 
-!!! warning
+Example:
+```julia
+    setprecision(256)
+```
+
+!!! note
+    Temporarily sets the precision for BigFloat operations to `precision` bits within the
+    scope of function `f`. This method is thread-safe due to the use of ScopedValue, ensuring
+    that precision settings do not interfere across threads.
 
-    This function is not thread-safe. It will affect code running on all threads, but
-    its behavior is undefined if called concurrently with computations that use the
-    setting.
+!!! compat "Julia 1.12"
+    Thread safety via ScopedValue is guaranteed starting with Julia 1.12. ([#51362])
```

Comment:
```suggestion
    Thread safety via ScopedValue is guaranteed starting with Julia 1.12.
```
As far as I know, compatibility notes typically do not refer to the PRs that introduced the respective changes.

Reply:
Thank you for your review.